### PR TITLE
Convert Button to TS

### PR DESCRIPTION
**Pull request by kyleshike**
Closes: #1026

### CODE REVIEW

File: src/Button/Button.stories.tsx [hunk 1]
```diff
@@ -543,7 +543,7 @@ export const Loading = () => (
     <Button
       isLoading={boolean('isLoading', true)}
       leadingIcon={faFileAlt}
-      loadingText={text('loadingText')}
+      loadingText={text('loadingText', 'Loading...')}
```
Comment (kyleshike) [inline src/Button/Button.stories.tsx:546]: When converting these stories to TS, it seems that `text()` expects a second arg

File: src/IconButton/IconButton.stories.jsx [hunk 2]
```diff
@@ -41,5 +41,6 @@ export const Sizes = () => (
   <>
     <IconButton action="DELETE" size="sm" />
     <IconButton action="DELETE" size="md" />
+    <IconButton action="DELETE" size="lg" />
```
Comment (gabescarbrough) [inline src/IconButton/IconButton.stories.jsx:44]: Nice. Love to see all the sizes represented.

File: src/Button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  OUTLINE_DANGER = 'outline-danger',
+  OUTLINE_PRIMARY = 'outline-primary',
+  OUTLINE_WARNING = 'outline-warning',
+  OUTLINE_TRANSPARENT = 'outline-transparent',
+  PRIMARY = 'primary',
+  TRANSPARENT = 'transparent',
+  WARNING = 'warning',
+}
+
+export type ButtonProps = RBButtonProps & {
```
Comment (gabescarbrough) [inline src/Button/Button.tsx:36]: This is such a clean pattern for extending props from components used within yours

**Review by gabescarbrough — APPROVED** [review]
This is awesome! Always glad to see more components using TypeScript

File: src/Button/Button.tsx (hunk 3 above)
Comment (kyleshike) [reply src/Button/Button.tsx:36]: I know, it's so lovely :)

**Pull request merged**